### ISSUE TITLE
artifacts introduced in D3D12 games on some AMD hardware when gamescope isnt compositing (kernel 6.8+)

### ISSUE THREAD

**Issue by matte-schwartz** [ValveSoftware/gamescope#1293]
I have bisected a [range of commits](https://gist.github.com/matte-schwartz/5dfa154e4938860da809d963681cdaac) in gamescope between 3.14.2 and 3.14.3 that seems to have introduced display artifacting on the OLED Deck, particularly when calling up the side menus or invoking the cursor in D3D12 games within gamescope-session. This only affects kernels 6.8.0-rc and later. I'm including a video of the artifacting here on amd-staging-drm-next, which I was advised to try by Mario @ AMD, and gamescope master: 

https://github.com/ValveSoftware/gamescope/assets/136293710/96bdea03-f075-4938-963d-f1718568354a

I'd bisect further but everything I've tried in my bisect range leads to some cursed behavior:

![IMG_1335](https://github.com/ValveSoftware/gamescope/assets/136293710/1f2b1cd9-7b56-49e1-a0c0-71475175eb60)

The issue occurs using both upstream HDR (AMD_PRIVATE_COLOR)/gamescope master as well as downstream HDR (the AMD_COLOR_STEAMDECK naming)/jupiter-3.6 gamescope. This seems to be separate from the issue that was present in 6.5.0-valve3 that had occasional display corruption, which was solved by: https://patchwork.freedesktop.org/patch/578255/ 

as a summary of my testing:

- gamescope built from the current head + kernels below 6.8 have no artifacting
- gamescope built from the current head + kernels above 6.8 have artifacting
- gamescope v3.14.2 + kernels above 6.8 have no artifacting
- gamescope v.3.14.3 and later + kernels above 6.8 have artifacting

I am adding a link to the original post I made for this issue on drm/amd, as I initially thought this was strictly tied to the kernel and not both the kernel and gamescope: https://gitlab.freedesktop.org/drm/amd/-/issues/3361

**Repro steps on my end while on 6.8 and later and latest gamescope:**

1. open a d3d12 game while in game mode
2. let it launch
3. spam toggle the quick menus a few times
4. hold the Steam button and use the right touchpad as a cursor and move the mouse a few times
5. let go of the Steam button and use the touchpad

**Comment by MCPO-Spartan-117** [comment]
That 'cursed behavior' is likely because you went before 7ed173f385a8a0c345dae256d523c699f51cd2c6 which is supposed to fix ce1156af053239e662aa021dbfc6a64cb770f64a, you can try to use `RADV_DEBUG=zerovram` to continue testing before 7ed173f385a8a0c345dae256d523c699f51cd2c6, may even fix the issue you are having.

**Comment by matte-schwartz** [comment]
closing as i've been unable to repro since 3.14.15. will reopen if it reappears.

**Comment by matte-schwartz** [comment]
I'll be putting together a more comprehensive summary of my findings tomorrow along with videos to showcase this more easily, but I've confirmed that this

1. affects all RDNA3 devices I own, (I do not own other generations except for the Steam Deck to test)
2. is still an issue on latest gamescope from https://github.com/ValveSoftware/gamescope/commit/09cb7b451d8825e597d15ab9d30e43aee8418670
3. is triggered by the "highest" modeset of each display I've tested (which gamescope defaults to)
4. is solved entirely by "Force composite" within gamescope-session, and is partially linked with https://github.com/ValveSoftware/gamescope/issues/1343 (forced compositing also fixes this)
5. is an AMD kernel regression on all kernels past 6.8, including `6.8.0-valve1-preview` and `amd-staging-drm-next` built from [c47ff6076a515378b4e85705b58b1e6685e57ae1](https://gitlab.freedesktop.org/agd5f/linux/-/commit/c47ff6076a515378b4e85705b58b1e6685e57ae1)
6. happens with all display connection types, including eDP (ROG Ally)
7. does **not** happen with nvidia
